src/functions.php

```diff
@@ -110,6 +110,40 @@ function is_first_key_operator($document)
     return (isset($firstKey[0]) && $firstKey[0] === '$');
 }
 
+/**
+ * Returns whether an update specification is a valid aggregation pipeline.
+ *
+ * @internal
+ * @param mixed $pipeline
+ * @return boolean
+ */
+function is_pipeline($pipeline)
+{
+    if (! is_array($pipeline)) {
+        return false;
+    }
+
+    if ($pipeline === []) {
+        return false;
+    }
+
+    foreach ($pipeline as $stage) {
```

Comment:
Should we also check that the array's keys follow a continuous sequence from zero like `_mongoc_document_is_pipeline()`? Otherwise, I think this function might return `true` for an associative array (nevermind numeric) consisting of a single DBRef (e.g. `[ 'x' => [ '$ref' => 'foo', '$id' => 'bar' ]]`). That seems like a bug.

I may be wrong, but either way I think it'd be worth adding some unit tests in FunctionsTest.php.

Reply:
Good point. I've added checks for the expected key and also added this specific example as a test (among others).